CRVResponse/src/MakeCrvWaveforms.cc

```diff
@@ -63,14 +63,15 @@ void MakeCrvWaveforms::MakeWaveform(const std::vector<double> &times,
   {
     double timeOfCharge=*iterTime;  //the time when the charge happened
     double charge=*iterCharge/_singlePEReferenceCharge;  //scale it to the 1PE reference charge used for the single PE waveform
-    if(timeOfCharge<startTime) continue; //this shouldn't happen
-    size_t waveformIndex = ceil((timeOfCharge-startTime)/digitizationPrecision);  //waveform index of the first digitization point for this particular charge
+    double waveformIndexTmp = ceil((timeOfCharge-startTime)/digitizationPrecision);
+    if(waveformIndexTmp<0) waveformIndexTmp=0;
+    size_t waveformIndex = static_cast<size_t>(lrint(waveformIndexTmp));  //waveform index of the first digitization point for this particular charge
     double waveformTime = waveformIndex*digitizationPrecision + startTime;  //the time for this waveform index
 
     for(; ; waveformIndex++, waveformTime+=digitizationPrecision)
     {
       double singlePEWaveformTime = waveformTime - timeOfCharge;
-      if(singlePEWaveformTime<0) continue;  //this shouldn't happen
```

Comment:
IIRC the previous conversation concluded that this should be a throw.   Did the changes above change this conclusion?

Reply:
The previous conversation was about another line, which I changed over the weekend. This line should not throw: There could be a very rare situation (which I have not observed in my test) that the singlePEWaveformTime is <0 due to precision issues when dealing with floating point numbers (after digitizing the time stamps and converting them back to floating point numbers). If this situation happens, the first point of the single PE waveform will be skipped, which will not result in a noticeable change of the total waveform. That's what the "continue" does.